src/task-actions.ts

```diff
@@ -88,8 +88,11 @@ export async function starknetCompileAction(args: TaskArguments, hre: HardhatRun
         });
     }
     if (hre.config.paths.cairoPaths) {
-        hre.config.paths.cairoPaths.forEach((path: string) => {
-            cairoPaths.push(path);
+        hre.config.paths.cairoPaths.forEach((cairPath: string) => {
+            if (!path.isAbsolute(cairPath)) {
```

Comment:
This check and the "fixing" of the path in the next line, should be done for all members of the final `cairoPaths` array. But looking at L98, we already do that. So I'm puzzled why this failed in the first place. Therefore we should introduce a test for this (probably can be a simple one, without the need to extend the example repo). We should first see if the test fails without your fix, and then if it passes with the fix.

Reply:
So, my fix was actually wrong. The issue only happens for a dockerized environment when we attempt to mount a non-absolute Cairo path. https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/307/commits/544ef8035f11adeecd134321baf27bfd08e6e668